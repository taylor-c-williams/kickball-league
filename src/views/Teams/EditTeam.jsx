import { useState, useEffect } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import EditForm from '../../components/Teams/TeamEditForm';
import { updateTeamById, getTeamById } from '../../services/teams';

export default function EditTeam() {
	const { id } = useParams();
	const [team, setTeam] = useState(null);
	const [name, setName] = useState('');
	const [city, setCity] = useState('');
	const [state, setState] = useState('');
	const [loading, setLoading] = useState(true);
	const history = useHistory();

	useEffect(() => {
		getTeamById(id)
			.then((res) => {
				setTeam(res);
				setName(res.name);
				setCity(res.city);
				setState(res.state);
			})
			.finally(() => setLoading(false));
	}, [id, setTeam]);

	if (loading) return <span className="loading">Loading...</span>;

	const handleSubmit = async (e) => {
		e.preventDefault();
		const res = await updateTeamById(id, { name, city, state });
		history.push(`/teams/${res[0].id}`);
	};
	return (
		<div>
			<h1>Edit Team</h1>
			{team.name}
			<EditForm
				name={name}
				city={city}
				state={state}
				handleSubmit={handleSubmit}
				setName={setName}
				setCity={setCity}
				setState={setState}
			/>
		</div>
	);
}
