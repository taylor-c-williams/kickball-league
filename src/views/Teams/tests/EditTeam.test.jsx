import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import EditTeam from '../EditTeam';

it('should render accurately the edit team view and form component', async () => {
	const container = render(
		<MemoryRouter initialEntries={['/editTeam/1']}>
			<Route path="/editTeam/:id">
				<EditTeam />
			</Route>
		</MemoryRouter>
	);

	const textEntry = await screen.findByText('Edit stuff', {
		exact: false,
	});

	expect(container).toMatchSnapshot();
	expect(textEntry).toBeInTheDocument();
});
