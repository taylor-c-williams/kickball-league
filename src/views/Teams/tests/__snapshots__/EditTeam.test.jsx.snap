// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render accurately the edit team view and form component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          Edit Team
        </h1>
        <div>
          <h1>
            chummy
          </h1>
          <fieldset>
            <legend>
              Edit stuff
            </legend>
            <form>
              <label
                for="name"
              >
                Team Name:
              </label>
              <input
                id="name"
                name="name"
                required=""
                type="text"
                value="chummy"
              />
              <label
                for="city"
              >
                City:
              </label>
              <input
                id="city"
                name="city"
                required=""
                type="text"
                value="bongo"
              />
              <label
                for="state"
              >
                State:
              </label>
              <input
                id="state"
                name="state"
                required=""
                type="text"
                value="bongo"
              />
              <input
                aria-label="Add New Team"
                type="submit"
                value="Update Team"
              />
            </form>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        Edit Team
      </h1>
      <div>
        <h1>
          chummy
        </h1>
        <fieldset>
          <legend>
            Edit stuff
          </legend>
          <form>
            <label
              for="name"
            >
              Team Name:
            </label>
            <input
              id="name"
              name="name"
              required=""
              type="text"
              value="chummy"
            />
            <label
              for="city"
            >
              City:
            </label>
            <input
              id="city"
              name="city"
              required=""
              type="text"
              value="bongo"
            />
            <label
              for="state"
            >
              State:
            </label>
            <input
              id="state"
              name="state"
              required=""
              type="text"
              value="bongo"
            />
            <input
              aria-label="Add New Team"
              type="submit"
              value="Update Team"
            />
          </form>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
